tools/profiling/Profiler.php

```diff
@@ -0,0 +1,288 @@
+
+        return $size;
+    }
+
+    /**
+     * Get var data
+     *
+     * @param mixed $var
+     *
+     * @return string|object
+     */
+    private function getVarData($var)
+    {
+        if (is_object($var)) {
+            return $var;
+        }
+
+        return (string) $var;
+    }
+
+    /**
+     * Intercept hook and register its data
+     *
+     * @param string $hookName
+     * @param array $params
+     */
+    public function interceptHook(string $hookName, array $params)
+    {
+        if (empty($this->hooksPerfs[$hookName])) {
+            $this->hooksPerfs[$hookName] = [
+                'time' => 0,
+                'memory' => 0,
+                'modules' => [],
+            ];
+        }
+
+        $this->hooksPerfs[$hookName]['time'] += $params['time'];
+        $this->hooksPerfs[$hookName]['memory'] += $params['memory'];
+        $this->hooksPerfs[$hookName]['modules'][] = $params;
+        $this->totalHooksMemory += $params['memory'];
+        $this->totalHooksTime += $params['time'];
+    }
+
+    /**
+     * Intercept module
+     *
+     * @param array $params
+     */
+    public function interceptModule(array $params)
+    {
+        $this->modulesPerfs[] = $params;
+        $this->totalModulesTime += $params['time'];
+        $this->totalModulesMemory += $params['memory'];
+    }
+
+    /**
+     * Process all data such as Global vars and
+     * database queries
+     */
+    public function processData()
+    {
+        // Including a lot of files uses memory
+        foreach (get_included_files() as $file) {
+            $this->totalFilesize += filesize($file);
+        }
+
+        foreach ($GLOBALS as $key => $value) {
+            if ($key === 'GLOBALS') {
+                continue;
+            }
+            $this->totalGlobalVarSize += ($size = $this->getVarSize($value));
+
+            if ($size > 1024) {
+                $this->globalVarSize[$key] = round($size / 1024);
+            }
+        }
+
+        arsort($this->globalVarSize);
+
+        $cache = Cache::retrieveAll();
+        $this->totalCacheSize = $this->getVarSize($cache);
+
+        // Sum querying time
+        $queries = Db::getInstance()->queries;
+        uasort($queries, [$this, 'sortByQueryTime']);
+        foreach ($queries as $data) {
+            $this->totalQueryTime += $data['time'];
+
+            $queryRow = [
+                'time' => $data['time'],
+                'query' => $data['query'],
+                'location' => str_replace('\\', '/', substr($data['stack'][0]['file'], strlen(_PS_ROOT_DIR_))) . ':' . $data['stack'][0]['line'],
+                'filesort' => false,
+                'rows' => 1,
+                'group_by' => false,
+                'stack' => [],
+            ];
+
+            if (preg_match('/^\s*select\s+/i', $data['query'])) {
+                $explain = Db::getInstance()->executeS('explain ' . $data['query']);
+                if (stristr($explain[0]['Extra'], 'filesort')) {
```

Comment:
```suggestion
                if (isset($explain[0]['Extra']) && stristr($explain[0]['Extra'], 'filesort')) {
```

In my shop a query has the content of `Extra` as `null`

Reply:
Yes it can, be the stristr will return false :thinking:

Reply:
> 
> 
> Yes it can, be the stristr will return false 🤔

It generates an error 500: `Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Type error: stristr() expects parameter 1 to be string, null given" at /tools/profiling/Profiler.php line 212`

![image](https://user-images.githubusercontent.com/49748656/92885201-d1858a80-f412-11ea-8d88-ea7d10d9fe58.png)

I dump `$explain[0]` and tells me that `Extra` is null: 
![image](https://user-images.githubusercontent.com/49748656/92886283-d139bf00-f413-11ea-974b-dc3475dd356f.png)

What would be the solution?

Reply:
Ohh you are right, I misread your comment and thought the issue was already in the code :sweat_smile: